### PR TITLE
Feature/basic file retrieval from git

### PR DESCRIPTION
**Pull request by jimCresswell**
Lots of changes, see individual commit messages for details.

Highlights:
* On the front page enter a HTTPS URL for a Git repo with feature files to see those feature files.

### CODE REVIEW

File: features-support/hooks/before.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+"use strict";
+/* eslint new-cap: 0 */
+
+module.exports = function beforeHooks() {
+
+  // Remove any old test data.
+  this.Before(function(callback) {
```
Comment (sponte) [inline features-support/hooks/before.js]: Would this not run before each test? I would only do it once before running a test suite

File: features-support/step_definitions/features.js [hunk 2]
```diff
@@ -1,44 +1,33 @@
...
-  // Remove old files.
-  return fs.removeTree(paths.public)
-    .catch(function(err) {
-      // Ignore failure to unlink missing directory.
-      if (err.code !== 'ENOENT') {
-        throw err;
+// Test helper.
+function getFeaturesFromUrl(callback) {
+  var world = this;
+  var projectFeaturesUrl = 'http://localhost:3000/?repo_url=' + encodeURIComponent(world.repoUrl);
```
Comment (sponte) [inline features-support/step_definitions/features.js]: would be better to parametrise port

File: features-support/step_definitions/features.js [hunk 3]
```diff
@@ -106,12 +96,19 @@ module.exports = function () {
...
     should.equal(this.statusCode, 200, "Bad HTTP status code.");
-    should.equal(/feature/i.test(this.body),
+    should.equal(/feature:/i.test(this.body),
       true,
       "The returned document body does not contain the word 'feature'");
     callback();
   });
 
+  this.Given(/^a URL representing a remote Git repo\.?$/, function (callback) {
+    this.repoUrl = "https://github.com/oss-specs/specs";
```
Comment (sponte) [inline features-support/step_definitions/features.js]: maybe accept specs url as a step argument, we'll be able to swap it out easily. Or leave it and change it when needed, up to you

File: features-support/hooks/before.js (hunk 1 above)
Comment (jimCresswell) [inline features-support/hooks/before.js]: If it doesn't run before every test then state can persist between states.

File: features-support/step_definitions/features.js (hunk 2 above)
Comment (jimCresswell) [inline features-support/step_definitions/features.js]: Done.

File: features-support/step_definitions/features.js (hunk 3 above)
Comment (jimCresswell) [inline features-support/step_definitions/features.js]: Done.

File: features-support/hooks/before.js (hunk 1 above)
Comment (sponte) [inline features-support/hooks/before.js]: This will become problematic when we want to run tests in parallel in the future

**Comment by jimCresswell** [comment]
Well I reverted the pull request so I could add changes to the test data handling, but annoyingly it doesn't reopen the pull request.

**Pull request merged**